File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/CustomerInfo.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{
    /// <summary>
    /// Used for Customer related information.
    /// </summary>
    /// <remarks>Use this class to set the customer related 
    /// information.</remarks>
    /// <example>
    /// <code lang="C#" escaped="false">
    ///	.................
    ///	// Inv is the Invoice object
    ///	.................
    ///	// Set the Customer Info details.
    ///	CustomerInfo Cust = new CustomerInfo();
    ///	Cust.CustCode = "CustXXXXX";
    ///	Cust.CustIP = "255.255.255.255";
    ///	Inv.CustomerInfo = Cust;
    ///	.................
    ///	</code>
    /// <code lang="Visual Basic" escaped="false">
    ///	.................
    ///	' Inv is the Invoice object
    ///	.................
    ///	' Set the Customer Info details.
    ///	Dim Cust As CustomerInfo = New CustomerInfo
    ///	Cust.CustCode = "CustXXXXX"
    ///	Cust.CustIP = "255.255.255.255"
    ///	Inv.CustomerInfo = Cust
    ///	.................
    ///	</code>
    /// </example>
    public sealed class CustomerInfo : BaseRequestDataObject
    {
        #region "Member Variables"

        /// <summary>
        /// Holds customer code
        /// </summary>
        private String mCustCode;

        /// <summary>
        /// Holds customer host name
        /// </summary>
        private String mCustHostName;

        /// <summary>
        /// Holds customer browser
        /// </summary>
        private String mCustBrowser;

        /// <summary>
        /// Holds customer IP
        /// </summary>
        private String mCustIP;

        /// <summary>
        /// Holds Customer Vat registration number
        /// </summary>
        private String mCustVatRegNum;

        /// <summary>
        /// Holds Customer's date of birth
        /// </summary>
        private String mDob;

        /// <summary>
        /// Holds customer id
        /// </summary>
        private String mCustId;

        /// <summary>
        /// Holds ReqName
        /// </summary>
        private String mReqName;

        /// <summary>
        /// Holds Customer Data
        /// </summary>
        private String mCustData;

        /// <summary>
        /// Holds Customer Id
        /// </summary>
        private String mCustomerId;

        /// <summary>
        /// Holds Customer Number
        /// </summary>
        private String mCustomerNumber;

        #endregion

        #region "Constructor"
        /// <summary>
        /// Constructor
        /// </summary>
        /// <remarks>Use this class to set the customer related 
        /// information.</remarks>
        /// <example>
        /// <code lang="C#" escaped="false">
        ///	.................
        ///	// Inv is the Invoice object
        ///	.................
        ///	// Set the Customer Info details.
        ///	CustomerInfo Cust = New CustomerInfo();
        ///	Cust.CustCode = "CustXXXXX";
        ///	Cust.CustIP = "255.255.255.255";
        ///	Inv.CustomerInfo = Cust;
        ///	.................
        ///	</code>
        /// <code lang="Visual Basic" escaped="false">
        ///	.................
        ///	' Inv is the Invoice object
        ///	.................
        ///	' Set the Customer Info details.
        ///	Dim Cust As CustomerInfo = New CustomerInfo
        ///	Cust.CustCode = "CustXXXXX"
        ///	Cust.CustIP = "255.255.255.255"
        ///	Inv.CustomerInfo = Cust
        ///	.................
        ///	</code>
        /// </example>
        public CustomerInfo()
        {
        }
        #endregion

        #region "Properties"

        /// <summary>
        /// Gets, Sets ReqName.
        /// </summary>
        /// <remarks>
        /// <para>Requester Name.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>REQNAME</code>
        /// </remarks>
        public String ReqName
        {
            get { return mReqName; }
            set { mReqName = value; }
        }

        /// <summary>
        /// Gets, Sets  CustCode.
        /// </summary>
        /// <remarks>
        /// <para>Customer code/customer reference ID.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>CUSTCODE</code>
        /// </remarks>
        public String CustCode
        {
            get { return mCustCode; }
            set { mCustCode = value; }
        }

        /// <summary>
        /// Gets, Sets  CustIP.
        /// </summary>
        /// <remarks>
        /// <para>Customer's IP address.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>CUSTIP</code>
        /// </remarks>
        public String CustIP
        {
            get { return mCustIP; }
            set { mCustIP = value; }
        }

        /// <summary>
        /// Gets, Sets  CustHostName.
        /// </summary>
        /// <remarks>
        /// <para>Customer's name of server that the account holder is connected to.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>CUSTHOSTNAME</code>
        /// </remarks>
        public String CustHostName
        {
            get { return mCustHostName; }
            set { mCustHostName = value; }
        }

        /// <summary>
        /// Gets, Sets  CustBrowser.
        /// </summary>
        /// <remarks>
        /// <para>Account holder’s HTTP browser type. Example:
        /// MOZILLA/4.0~(COMPATIBLE;~MSIE~5.0;~WINDOWS~95)</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>CUSTBROWSER</code>
        /// </remarks>
        public String CustBrowser
        {
            get { return mCustBrowser; }
            set { mCustBrowser = value; }
        }

        /// <summary>
        /// Gets, Sets  CustVatRegNum.
        /// </summary>
        /// <remarks>
        /// <para>Customer's VAT registrations number.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>CUSTVATREGNUM</code>
        /// </remarks>
        public String CustVatRegNum
        {
            get { return mCustVatRegNum; }
            set { mCustVatRegNum = value; }
        }

        /// <summary>
        /// Gets, Sets  DOB.
        /// </summary>
        /// <remarks>
        /// <para>Account holder’s date of birth.</para>
        /// <para>Format: mmddyyyy.</para>
        /// <para>mm - Month, dd - Day, yy - Year.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>DOB</code>
        /// </remarks>
        public String DOB
        {
            get { return mDob; }
            set { mDob = value; }
        }


        /// <summary>
        /// Gets, Sets  CustId.
        /// </summary>
        /// <remarks>
        /// <para>Customer's Id.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>CUSTID</code>
        /// </remarks>
        public String CustId
        {
            get { return mCustId; }
            set { mCustId = value; }
        }

                /// <summary>
        /// Gets, Sets Customer Data
        /// </summary>
        /// <remarks>
        /// <para>Requester Name.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>CUSTDATAE</code>
        /// </remarks>
        public String CustData
        {
            get { return mCustData; }
            set { mCustData = value; }
        }

        /// <summary>
        /// Gets, Sets Customer identification.
        /// </summary>
        /// <remarks>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>CUSTOMERID</code>
        /// </remarks>
        public String CustomerId
        {
            get { return mCustomerId; }
            set { mCustomerId = value; }
        }

        /// <summary>
        /// Gets, Sets Customer Number.
        /// </summary>
        /// <remarks>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>CUSTOMERNUMBER</code>
        /// </remarks>
        public String CustomerNumber
        {
            get { return mCustomerNumber; }
            set { mCustomerNumber = value; }
        }
        #endregion

        #region "Core functions"

        /// <summary>
        /// Generates the transaction request.
        /// </summary>
        internal override void GenerateRequest()
        {
            try
            {
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_REQNAME, mReqName));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTCODE, mCustCode));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTIP, mCustIP));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTVATREGNUM, mCustVatRegNum));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_DOB, mDob));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTID, mCustId));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTHOSTNAME, mCustHostName));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTBROWSER, mCustBrowser));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTDATA, mCustData));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTOMERID, mCustomerId));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CUSTOMERNUMBER, mCustomerNumber));
            }
            catch (BaseException)
            {
                throw;
            }
            catch (Exception Ex)
            {
                DataObjectException DEx = new DataObjectException(Ex);
                throw DEx;
            }
            //catch
            //{
            //    throw new Exception();
            //}
        }

        #endregion
    }
}

File: java/src/sdk/dataobjects/paypal/payflow/CustomerInfo.java
package paypal.payflow;

/**
 * Used for Customer related information.
 * <p>Use this class to set the customer related
 * information.</p>
 *
 *  .................
 * // inv is the Invoice object
 * .................
 * // Set the Customer Info details.
 * CustomerInfo cust = new CustomerInfo();
 * cust.setCustCode ("CustXXXXX");
 * cust.setCustIP ("255.255.255.255");
 * inv.setCustomerInfo (cust);
 * .................
 */
public final class CustomerInfo extends BaseRequestDataObject {

    private String custCode;
    private String custIP;
    private String custVatRegNum;
    private String dob;
    private String custId;
    private String reqName;
    private String custHostName;
    private String custBrowser;

    /**
     * Constructor
     * <p>Use this class to set the customer related
     * information.</p>
     */
    public CustomerInfo() {
    }

    protected void generateRequest() {
        try {
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_REQNAME, reqName));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CUSTCODE, custCode));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CUSTIP, custIP));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CUSTVATREGNUM, custVatRegNum));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_DOB, dob));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CUSTID, custId));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CUSTHOSTNAME, custHostName));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CUSTBROWSER, custBrowser));
        } catch (Exception ex) {
            ErrorObject err = new ErrorObject(PayflowConstants.SEVERITY_FATAL, "", ex.toString());
            if (getContext() != null) {
                getContext().addError(err);
            }
        }
    }

    /**
     * Gets the CustCode
     * <p>Customer code/customer reference ID.</P>
     *
     * @return custCode String
     *  <p>Maps to Payflow Parameter: CUSTCODE</p>
     */

    public String getCustCode() {
        return custCode;
    }

    /**
     * Sets the CustCode.
     * <p>Customer code/customer reference ID.</P>
     *
     * @param custCode String
     *  <p>Maps to Payflow Parameter: CUSTCODE</p>
     */
    public void setCustCode(String custCode) {
        this.custCode = custCode;
    }

    /**
     * Gets the Customer's Id.
     *
     * @return custId String
     *  <p>Maps to Payflow Parameter: CUSTID</p>
     */

    public String getCustId() {
        return custId;
    }

    /**
     * Sets the Customer's Id.
     *
     * @param custId String
     *  <p>Maps to Payflow Parameter: CUSTID</p>
     */
    public void setCustId(String custId) {
        this.custId = custId;
    }

    /**
     * Gets the Customer's IP address.
     *
     * @return custIP String
     *  <p>Maps to Payflow Parameter: CUSTIP</p>
     */
    public String getCustIP() {
        return custIP;
    }

    /**
     * Sets the Customer's IP address.
     *
     * @param custIP String
     *  <p>Maps to Payflow Parameter: CUSTIP</p>
     */
    public void setCustIP(String custIP) {
        this.custIP = custIP;
    }

    /**
     * Gets the Customer's VAT registrations number.
     *
     * @return custVatRegNum String
     *  <p>Maps to Payflow Parameter: CUSTVATREGNUM</p>
     */
    public String getCustVatRegNum() {
        return custVatRegNum;
    }

    /**
     * Sets the Customer's VAT registrations number.
     *
     * @param custVatRegNum String
     *  <p>Maps to Payflow Parameter: CUSTVATREGNUM</p>
     */

    public void setCustVatRegNum(String custVatRegNum) {
        this.custVatRegNum = custVatRegNum;
    }

    /**
     * Gets the dob.
     * <p>Account holder's date of birth.</p>
     * <p>Format: mmddyyyy.</p>
     * <p>mm - Month, dd - Day, yy - Year.</P>
     *
     * @return String
     *  <p>Maps to Payflow Parameter: DOB</p>
     */
    public String getDob() {
        return dob;
    }

    /**
     * Sets the dob.
     * <p>Account holder's date of birth.</p>
     * <p>Format: mmddyyyy.</p>
     * <p>mm - Month, dd - Day, yy - Year.</P>
     *
     * @param dob String
     *  <p>Maps to Payflow Parameter: DOB</p>
     */
    public void setDob(String dob) {
        this.dob = dob;
    }

    /**
     * Gets the Requester Name.
     * @return String
     *
     * <p>Maps to Payflow Parameter: REQNAME</p>
     */
    /**
     * @return Returns the reqName.
     */
    public String getReqName() {
        return reqName;
    }

    /**
     * Sets the Requester Name.
     * @param String
     *
     * <p>Maps to Payflow Parameter: REQNAME</p>
     */
    /**
     * @param reqName The reqName to set.
     */
    public void setReqName(String reqName) {
        this.reqName = reqName;
    }

    /**
     * Gets the Customer's Host Name.
     * @return String
     *
     * <p>Maps to Payflow Parameter: CUSTHOMENAME</p>
     */
    /**
     * @return Returns the custHostName.
     */
    public String getCustHostName() {
        return custHostName;
    }

    /**
     * Sets the Customer's Host Name.
     * @param String
     *
     * <p>Maps to Payflow Parameter: CUSTHOMENAME</p>
     */
    /**
     * @param custHostName The customer host name to set.
     */
    public void setCustHostName(String custHostName) {
        this.custHostName = custHostName;
    }

    /**
     * Gets the Customer's Browser.
     * @return String
     *
     * <p>Maps to Payflow Parameter: CUSTBROWSER</p>
     */
    /**
     * @return Returns the custBrowser.
     */
    public String getCustBrowser() {
        return custBrowser;
    }

    /**
     * Sets the Customer's Browser.
     * @param String
     *
     * <p>Maps to Payflow Parameter: CUSTBROWSER</p>
     */
    /**
     * @param custBrowser The customers browser.
     */
    public void setCustBrowser(String custBrowser) {
        this.custBrowser = custBrowser;
    }
}
